framework/base/Behavior.php

```diff
@@ -85,8 +85,16 @@ public function attach($owner)
     public function detach()
     {
         if ($this->owner) {
-            foreach ($this->events() as $event => $handler) {
-                $this->owner->off($event, is_string($handler) ? [$this, $handler] : $handler);
+            foreach ($this->events() as $event => $value) {
+                if (is_string($value)) {
+                    $handler = [$this, $value];
+                } elseif ($value instanceof \Closure) {
+                    $handler = null;
```

Comment:
This will also remove other event handlers.

Are you sure that it will be correct to remove a event completely? 
What if the event is ActiveRecord::EVENT_AFTER_FIND?

Reply:
I thought a lot about this problem, but I didn't find a solution. We just can't distinct one Closure from another one. That is why we can't remove one specific handler from event list.